src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -91,18 +175,14 @@ public String getTimeDateFunctions() throws SQLException {
     @Override
     public ResultSet getProcedures(
             String catalog, String schemaPattern, String procedureNamePattern) throws SQLException {
-        MongoJsonSchema resultSchema = MongoJsonSchema.createEmptyObjectSchema();
-        resultSchema.addRequiredScalarKeys(
-                new Pair<>(PROCEDURE_CAT, BsonTypeInfo.STRING_TYPE_NAME),
-                new Pair<>(PROCEDURE_SCHEM, BsonTypeInfo.STRING_TYPE_NAME),
-                new Pair<>(PROCEDURE_NAME, BsonTypeInfo.STRING_TYPE_NAME),
-                new Pair<>(REMARKS, BsonTypeInfo.STRING_TYPE_NAME),
-                new Pair<>(PROCEDURE_TYPE, BsonTypeInfo.INT_TYPE_NAME),
-                new Pair<>(SPECIFIC_NAME, BsonTypeInfo.STRING_TYPE_NAME));
-
-        // All fields in this result set are nested under the bottom namespace.
-        MongoJsonSchema botSchema = MongoJsonSchema.createEmptyObjectSchema();
-        botSchema.properties.put(BOT_NAME, resultSchema);
```

Comment:
All schema creation code followed this exact same pattern, so I extracted all of that into the `createBottomSchema` helper method.